docs/registry/index.rst

```diff
@@ -276,29 +276,190 @@ For an example of how this might look like, see `GAVO's plate tutorial`_
 
 .. _GAVO's plate tutorial: http://docs.g-vo.org/gavo_plates.pdf
 
+More examples
+-------------
+
+Discover archives
+^^^^^^^^^^^^^^^^^
+
+You can use the registry ``search`` method (or the ``regsearch`` function) 
+to discover archives that may have x-ray images and then query those archives 
+to find what x-ray images that have of CasA. For the arguments you will 
+enter ``'image'`` for the service type and ``'x-ray'`` for the waveband. 
+The position is provided by the Astropy library.
+
+The query returns a :py:class:`~pyvo.registry.regtap.RegistryResults` object 
+which is a container holding a table of matching services. In this example 
+it returns 33 matching services.
+
+.. doctest-remote-data::
+
+  >>> import pyvo as vo
+  >>> from astropy.coordinates import SkyCoord
+  >>>
+  >>> import warnings
+  >>> warnings.filterwarnings('ignore', module="astropy.io.votable.*")
+  >>>
+  >>> archives = vo.regsearch(servicetype='image', waveband='x-ray') 
+  >>> pos = SkyCoord.from_name('Cas A')
+  >>> len(archives)   # doctest: +IGNORE_OUTPUT
+  33
+
+There are also other type of services that you can choose via the 
+``servicetype`` parameter:
+
+  ===== ======================================
+  sia   Simple Image Access (SIA) services
+  ssa   Simple Spectral Access (SSA) services
+  scs   Simple Cone Search (SCS) services
+  slap  Simple Line Access (SLA) services
+  tap   Table Access Protocol (TAP) services
+  ===== ======================================
+
+You can learn more about the archives by printing their titles 
+and access URL:
+
+.. doctest-remote-data::
+
+  >>> for service in archives:    # doctest: +ELLIPSIS
+  ...     print(service.res_title, service.access_url)
+  Chandra X-ray Observatory Data Archive https://cda.harvard.edu/cxcsiap/queryImages?
+  Chandra Source Catalog http://cda.cfa.harvard.edu/cscsiap/queryImages?
+  Chandra Source Catalog Release 1 http://cda.cfa.harvard.edu/csc1siap/queryImages?
+  ...
+
+It is not neccessary to keep track of the URL because you can search 
+images directly from the registry record, for example using the Chandra 
+X-ray Observatory (CDA) service and the ``search`` method, inserting 
+the position and size for the desired object.
+
+.. doctest-remote-data::
+
+  >>> images = archives["CDA"].search(pos=pos, size=0.25)
+  >>> len(images)   # doctest: +IGNORE_OUTPUT
+  822
+
+Sometimes you are looking for a type of object. For this purpose, the  
+``keywords`` parameter is useful here. For example, you want to find 
+all catalogs related to blazars observed with Fermi:
+
+.. doctest-remote-data::
+
+  >>> cats = vo.regsearch(keywords=['blazar','Fermi'])
```

Comment:
Some style fixes are needed for the example, nothing serious, but adding and removing a few whitespaces here and there. I'm happy to do such fixes before merging.

Reply:
I am very sorry for replying after such a long time, I have not been focusing on GitHub for a while. I have made changes such as:
1. use the `~pyvo.registry.Servicetype` link for the parameters
2. some whitespaces fixes.
For this one:

.. doctest-remote-data::

  >>> for service in archives:    # doctest: +ELLIPSIS

I used `ELLIPSIS` because the output is a very long list of services, I just took the first three of them as examples. I don't think without `ELLIPSIS` it would pass the test.

And... I tried to rebase/squash all commits into one, and somehow the result does not look like what I wanted. But the overall changes are still there, I would appreciate if you could help me here. :)

Reply:
Re rebase, here is the  big picture and the copy paste of what I see in my local setup:

1) locally you need the clone of astropy, and your fork, check it with `git remote -v`
2) fetch to update the remotes
3) rebase
4) push back to your remote branch.

```
$ git remote -v
astropy	git@github.com:astropy/pyvo (fetch)
astropy	git@github.com:astropy/pyvo (push)
bsipocz	git@github.com:bsipocz/pyvo (fetch)
bsipocz	git@github.com:bsipocz/pyvo (push)
chmmao	git@github.com:chmmao/pyvo (fetch)
chmmao	git@github.com:chmmao/pyvo (push)
$ git fetch --all --prune
Fetching bsipocz
Fetching astropy
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (1/1), 660 bytes | 660.00 KiB/s, done.
From github.com:astropy/pyvo
   b96a950..feda0f0  main       -> astropy/main
Fetching chmmao
$ git rebase -i astropy/main
# editor comes up, do the commit reorders, squashes, removal of duplicates (if there is any from a previous rebase, etc). Note that we rebase on astropy/main and not on the main of your fork
# 
# When rebase is finished (you may need to do a bunch of git rebase --continue, etc., push it back. 
$ git push chmmao HEAD:integrate_examples --force
# note about the --force, it's needed as you overwrite the branch
```